auctions/dutch-auction/project/tests/functions/auction.rs

```diff
@@ -0,0 +1,214 @@
+mod passing {
+    use crate::utils::{
+        auction, bid, cancel_auction, change_asset, change_beneficiary, create_auction,
+        get_contract_instance,
+    };
+    use fuels::{
+        signers::WalletUnlocked,
+        types::{ContractId, Identity},
+    };
+
+    #[tokio::test]
+    async fn can_get_auction_details() {
+        let (instance, wallet) = get_contract_instance().await;
+
+        create_auction(
+            &instance,
+            400,
+            100,
+            2, // Block height will be 1 at start, then 2 during sending of this tx
+            5,
+            Identity::Address(wallet.address().into()),
+            ContractId::zeroed(),
+        )
+        .await;
+
+        let auc = auction(&instance, 1).await;
+
+        assert_eq!(auc.asset_id, ContractId::zeroed());
```

Comment:
You could look into the escrow which has `EscrowInfo` for events but something similar could be implemented here